libs/features/spectrum2-styles/build.mjs

```diff
@@ -0,0 +1,84 @@
+import fs from 'fs';
+import path from 'path';
+import postcss from 'postcss';
+
+const spectrumCSSPath = path.resolve('node_modules/@spectrum-css/tokens/dist/index.css');
+const miloCSSPath = path.resolve('../../deps/spectrum2-styles.css');
+
+const spectrumCSS = fs.readFileSync(spectrumCSSPath, 'utf8');
+const miloCSS = fs.readFileSync(miloCSSPath, 'utf8');
+
+// Formats and transforms Spectrum CSS Rgb and opacity properties for performance
+const transformRgbProperties = (rule) => {
+  const properties = {};
+
+  const replaceZeroPx = (value) => value.replace(/\b0px\b/g, '0');
+
+  rule.walkDecls((decl) => {
+    const { prop, value } = decl;
+
+    decl.value = replaceZeroPx(value);
+
+    if (prop.endsWith('-rgb')) {
+      const baseName = prop.replace('-rgb', '');
+      properties[baseName] = `rgb(${value.replace(/,\s*/g, ' ')})`;
+      decl.remove();
+    }
+
+    if (prop.endsWith('-opacity')) {
+      const baseName = prop.replace('-opacity', '');
+      properties[baseName] = properties[baseName] ? properties[baseName].replace(')', ` / ${parseFloat(value) * 100}%)`) : `rgb(0 0 0 / ${value})`;
+      decl.remove();
+    }
+  });
+
+  Object.keys(properties).forEach((prop) => rule.append({ prop, value: properties[prop] }));
+};
```

Comment:
I can't put my finger on it, but I think this can be simplified a bit. Since you write the rgb values & then re-write them for the opacity, it seems like a bit of redundant work.

Reply:
hmm, maybe you can explain this a little more? I'm not sure what you mean by 're-writing' the rgb values for opacity.
So to clarify a little bit here, we are receiving values like this from Spectrum CSS Tokens:
```
  --spectrum-transparent-white-900-rgb: 255, 255, 255;
  --spectrum-transparent-white-900-opacity: 0.94;
```

and we're transforming and combining these into a singe property like this, to format and reduce file size.

```
--s2-transparent-white-900: rgb(255 255 255 / 94%);
```
This code identifies `-rgb` appended css properties, transforms the value from the `255, 255, 255` format into `rgb( 255 255 255)` and sets it to the new `--s2-transparent-white-900` (un-appended base name) property in the properties obj.

It then identifies the `-opacity` appended css property, checks if there is an existing associated base name in the obj, if one exists it will add in the formatted opacity value ` / 94% )` to the formatted rgb value. So `--s2-transparent-white-900` now becomes `rgb(255 255 255 / 94%)`. It should fallback to `rgb(0 0 0 / ${value})` if there is no associated rgb value.

Let me know if this clarifies that further? Or if you think this can be improved in some way.

Reply:
I was thinking about something like this:

```js
rule.walkDecls((decl) => {
  const { prop, value } = decl;

  decl.value = replaceZeroPx(value);

  let baseName;
  let isRgb = prop.endsWith('-rgb');
  let isOpacity = prop.endsWith('-opacity');

  if (isRgb || isOpacity) {
    baseName = prop.replace(isRgb ? '-rgb' : '-opacity', '');
    properties[baseName] = isRgb ? `rgb(${value.replace(/,\s*/g, ' ')})` : 
      properties[baseName] ? properties[baseName].replace(')', ` / ${parseFloat(value) * 100}%)`) : `rgb(0 0 0 / ${value})`;
    decl.remove();
  }
});
```

Reply:
Nice! 👏🏼 .  I think milo's eslint has some opinions about nested ternary's I can adjust but I like this consolidation.